### PR TITLE
should solve #26 and more

### PR DESCRIPTION
**Pull request by SimonDanisch**
@mschauer if you have some time it'd be nice if you can see if its not
only locally solved ;)

### CODE REVIEW

**Comment by SimonDanisch** [comment]
I also didn't realize that `@inbounds` has an effect for ntuples.. But it seems it does, so I put it in a lot of places now :D 
By the way, https://github.com/johnmyleswhite/Benchmarks.jl is a nice package to benchmark stuff that's usually too fast to benchmark

**Comment by mschauer** [comment]
Yes, it works fine on this machine and with expm(::Mat{3,3,T})

```
julia> @time @repeat 10^5 eye(Mat{3,3,Float64}) + 2* expm(B)
  0.034894 seconds (400.00 k allocations: 30.518 MB, 4.13% gc time
```

PS: Thanks, Ill have a look at Benchmarks

File: src/core.jl [hunk 1]
```diff
@@ -20,23 +20,30 @@ function fixedsizearray_type{FSA <: FixedArray}(::Type{FSA})
...
-eltype{T <: FixedArray}(A::Type{T})                 = eltype(fixedsizearray_type(T))
-eltype{T <: FixedArray}(A::T)                       = eltype(T)
+eltype{T <: FixedArray}(A::Type{T})                 = eltype(super(T))
+eltype{T <: FixedArray,N,SZ}(A::FixedArray{T,N,SZ}) = T
 
-length{T,N,SZ}(A::Type{FixedArray{T,N,SZ}})         = prod(SZ.parameters)::Int
-length{T <: FixedArray}(A::Type{T})                 = length(fixedsizearray_type(T))
-length{C,T}(::FixedVector{C,T})                     = C
-length{R,C,T}(::FixedMatrix{R,C,T})                 = R*C
+_length{T <: Tuple}(::Type{T})						= *(SZ.parameters...)
```
Comment (mschauer) [inline src/core.jl]: SZ does not appear on the left hand side

File: src/core.jl [hunk 2]
```diff
@@ -20,23 +20,30 @@ function fixedsizearray_type{FSA <: FixedArray}(::Type{FSA})
...
 
 
 ndims{T,N,SZ}(A::Type{FixedArray{T,N,SZ}})          = N
-ndims{T <: FixedArray}(A::Type{T})                  = ndims(fixedsizearray_type(T))
+ndims{T <: FixedArray}(A::Type{T})                  = ndims(super(T))
 ndims{T <: FixedArray}(A::T)                        = ndims(T)
 
-size{T,N,SZ}(A::Type{FixedArray{T,N,SZ}})           = (SZ.parameters...)::NTuple{N, Int}
-size{T <: FixedArray}(A::Type{T})                   = size(fixedsizearray_type(T))
+_size{T <: Tuple}(::Type{T})						= (SZ.parameters...)
```
Comment (mschauer) [inline src/core.jl]: same

**Pull request merged**